GameData/RealismOverhaul/Waterfall_Configs/_Processor/18_late-initialize.cfg

```diff
@@ -0,0 +1,54 @@
+// Initialize optional keys that may be inserted by AutoConfig.
+
+@PART[*]:HAS[@ROWaterfall]:FOR[zROWaterfall_18_LateInitialize]:NEEDS[Waterfall]
+{
+    @ROWaterfall
+    {
+        @ExtraTemplate,*
+        {
+            // Required key: `template`.
+            &transform = #$../transform$
+            &position = #$../position$
+            &rotation = #$../rotation$
+            &scale = #$../scale$
+        }
+
+        @MainPlumeVariant,*
+        {
+            // Required key: `name`.
+            // Required key: `template`.
```

Comment:
I'm not sure how well "obviously incorrect value" would work here for `template`, because specifying a template that doesn't exist causes Waterfall to emit NREs at runtime. IMHO that's even more puzzling behavior.

Using a `MainPlumeVariant` requires you to manually specify which RFMEC configs use which `name`s. Thus, I don't think it's likely that people will accidentally omit this key.

FWIW, there's documentation in the body of the PR that specifies what keys are required.

Reply:
Provide that as a bug report to Waterfall: they shouldn't emit NREs and should instead detect the invalid config and message an appropriate error in the log.  Which would be much more clear when you see something like `ERR: ModuleWaterfallFX tried to load a template with name "ROWaterfall-SomeoneForgotToAssignATemplateName" but couldn't find one!`

Documentation's great.  Now if only it prevented people from writing broken patches, even folks who know what they're doing.